tlslite/integration/clienthelper.py

```diff
@@ -103,8 +104,14 @@ def __init__(self,
 
         self.tlsSession = None
 
-        if not self._isIP(host):
+        if host is not None and not self._isIP(host):
+            # remove port, if present
+            colon = host.find(':')
+            if colon > 0:
+                host = host[:colon]
             self.serverName = host
+            if host and not is_valid_hostname(host):
+                raise ValueError("Invalid hostnmae: {0}".format(host))
         else:
             self.serverName = None
```

Comment:
port from host is unused because the host is used only for SNI (that's why there's a check if it's an IP)

I've added docs to the missing fields (it's easier to fix them up little by little than to do a big rework anyway)